force-app/main/default/classes/SOQL.cls

```diff
@@ -288,20 +333,30 @@ public with sharing class SOQL {
      */
     public static List<Account> findAccountsByLeadEmailDomains(Set<Id> leadIds) {
         // Query the Leads
-
+        List<Lead> leadSet = [SELECT Id, LastName, Email FROM Lead WHERE Id = :leadIds];
         // Create a set of email domains from the Leads
-
+        Set<String> emailDomains = new Set<String>();
         // Loop through each Lead
-
-            // Get the email from the Lead
-
-            // Check if the email is not null
-
-                // Add the email domain to the set
-
-        
+        for(Lead l :leadSet){
+        // Get the email from the Lead
+        // Check if the email is not null
+            if(l.Email != null){
+                 // Add the email domain to the set
+                 String domain = l.Email.substring(l.Email.indexOf('@') + 1);
```

Comment:
Nicely done, Also try the substringAfter method.
Try to avoid single letter variables